### PR TITLE
FINERACT-822 enable OperatorPrecedence error

### PR DESCRIPTION
**Pull request by percyashu**
FINERACT-822

### CODE REVIEW

**Review by awasum — APPROVED** [review]
This LGTM. @xurror @ptuomola , @vorburger will you like to also take a closer look at this? and merge?

**Review by xurror — APPROVED** [review]
I don't know about this feature. Uh.., the compiler most certainly understands this stuff without the brackets, but on our end, readability can be tough with the bracket delimiters. But I'm OK merging this, it can't be a very big deal. Just resolve the merge conflicts and let me know when build passes.

**Comment by percyashu** [comment]
@xurror done

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/GlobalConfigurationProperty.java [hunk 1]
```diff
@@ -114,8 +114,8 @@ public Date getDateValue(){
 
         final String passwordPropertyName = "force-password-reset-days";
         if (this.name.equalsIgnoreCase(passwordPropertyName)) {
-            if (this.enabled == true && command.hasParameter(valueParamName) && this.value == 0 || this.enabled == true
-                    && !command.hasParameter(valueParamName) && previousValue == 0) { throw new ForcePasswordResetException(); }
+            if (((this.enabled == true) && command.hasParameter(valueParamName) && (this.value == 0)) || ((this.enabled == true)
+                    && !command.hasParameter(valueParamName) && (previousValue == 0))) { throw new ForcePasswordResetException(); }
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/GlobalConfigurationProperty.java]: Wait, this looks weird to me, shouldn't this just be:

```suggestion
            if ((this.enabled == true && command.hasParameter(valueParamName) && this.value == 0) || (this.enabled == true
                    && !command.hasParameter(valueParamName) && previousValue == 0)) { throw new ForcePasswordResetException(); }
```

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java [hunk 2]
```diff
@@ -2384,7 +2384,7 @@ public ChangedTransactionDetail disburse(final AppUser currentUser, final JsonCo
          **/
 
         if (isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()
-                        && (isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1 || !isMultiDisburmentLoan())) {
+                        && ((isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1) || !isMultiDisburmentLoan())) {
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java]: Wouldn't this be better & clearer like this, instead:

    if ((isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()
        && isMultiDisburmentLoan() && getDisbursedLoanDisbursementDetails().size() == 1) || !isMultiDisburmentLoan()) {

**Review by vorburger — CHANGES_REQUESTED** [review]

**Comment by xurror** [comment]
Still some merge conflicts and maybe also apply suggested changes.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java (hunk 2 above)
Comment (percyashu) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java]: OK will try this but the parentheses grouping might change the logic of the check

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/GlobalConfigurationProperty.java (hunk 1 above)
Comment (percyashu) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/domain/GlobalConfigurationProperty.java]: yeah will modify this

**Comment by xurror** [comment]
Good work.

**Pull request merged**